src/date.ts

```diff
@@ -121,7 +121,11 @@ const localMonth = (date: Date): string => months[date.getMonth()];
 
 function fromString(date: string): Option<Date> {
 	try {
-		return some(new Date(date));
+		const maybeDate = new Date(date);
+		if (!isValidDate(maybeDate)) {
+			throw new Error('not a valid date');
```

Comment:
I can't remember why this throw has been added .. does it matter if we're throwing anything here since we're catching it and returning a `none` anyway?

Reply:
we added this because the parameter could be an invalid date string like `(undefined as unknown) as string` or `'random'` but it still doesn't throw any exception and maybeDate gets a value of `Invalid Date`. And the result of the function would become some(Invalid Date)

I actually couldn't find any scenario that would throw an exception.